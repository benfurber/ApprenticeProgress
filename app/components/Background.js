import React, { Component } from "react";
import { Image, StyleSheet } from "react-native";
import { Container, View } from "native-base";

class Background extends Component {
  render() {
    const { children } = this.props;

    return (
      <Container style={styles.container}>
        <View style={styles.childrenView}>{children}</View>
        <View style={styles.imageView}>
          {this._renderImage()}
          {this._renderImage()}
          {this._renderImage()}
        </View>
        {this._renderRalph()}
      </Container>
    );
  }

  _renderImage() {
    return (
      <Image
        source={require("../assets/images/background.png")}
        resizeMode="repeat"
      />
    );
  }

  _renderRalph() {
    const { noRalph } = this.props;

    if (!noRalph) {
      return (
        <View style={styles.ralphView}>
          <Image
            source={require("../assets/images/ralph.png")}
            style={styles.ralph}
          />
        </View>
      );
    }
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: "stretch",
    flex: 1,
    flexDirection: "column-reverse",
    justifyContent: "flex-start",
  },
  childrenView: {
    flex: 1,
    zIndex: 2,
  },
  imageView: {
    flex: 1,
    position: "absolute",
    zIndex: 0,
  },
  ralphView: {
    right: -35,
    flexDirection: "row",
    position: "absolute",
    zIndex: 1,
  },
  ralph: {
    alignSelf: "flex-end",
  },
});

export { Background };
