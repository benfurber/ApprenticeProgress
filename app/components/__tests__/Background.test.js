import React from "react";
import { Text } from "react-native";
import { shallow } from "enzyme";

import { Background } from "../Background";

describe("Background", () => {
  it("has a valid snapshot", () => {
    const component = shallow(
      <Background>
        <Text>Hello</Text>
      </Background>
    );

    expect(component).toMatchSnapshot();
  });
});
