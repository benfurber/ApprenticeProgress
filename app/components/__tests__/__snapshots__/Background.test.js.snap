// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background has a valid snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Background>
    <Component>
      Hello
    </Component>
  </Background>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(ViewNB)
          style={
            Object {
              "flex": 1,
              "zIndex": 2,
            }
          }
        >
          <Component>
            Hello
          </Component>
        </Styled(ViewNB)>,
        <Styled(ViewNB)
          style={
            Object {
              "flex": 1,
              "position": "absolute",
              "zIndex": 0,
            }
          }
        >
          <Image
            resizeMode="repeat"
            source={
              Object {
                "testUri": "../../../app/assets/images/background.png",
              }
            }
          />
          <Image
            resizeMode="repeat"
            source={
              Object {
                "testUri": "../../../app/assets/images/background.png",
              }
            }
          />
          <Image
            resizeMode="repeat"
            source={
              Object {
                "testUri": "../../../app/assets/images/background.png",
              }
            }
          />
        </Styled(ViewNB)>,
        <Styled(ViewNB)
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "position": "absolute",
              "right": -50,
              "zIndex": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../app/assets/images/ralph.png",
              }
            }
            style={
              Object {
                "alignSelf": "flex-end",
                "flex": 1,
              }
            }
          />
        </Styled(ViewNB)>,
      ],
      "style": Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "column-reverse",
        "justifyContent": "flex-start",
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Component>
            Hello
          </Component>,
          "style": Object {
            "flex": 1,
            "zIndex": 2,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Hello",
          },
          "ref": null,
          "rendered": "Hello",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Image
              resizeMode="repeat"
              source={
                Object {
                  "testUri": "../../../app/assets/images/background.png",
                }
              }
            />,
            <Image
              resizeMode="repeat"
              source={
                Object {
                  "testUri": "../../../app/assets/images/background.png",
                }
              }
            />,
            <Image
              resizeMode="repeat"
              source={
                Object {
                  "testUri": "../../../app/assets/images/background.png",
                }
              }
            />,
          ],
          "style": Object {
            "flex": 1,
            "position": "absolute",
            "zIndex": 0,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "resizeMode": "repeat",
              "source": Object {
                "testUri": "../../../app/assets/images/background.png",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "resizeMode": "repeat",
              "source": Object {
                "testUri": "../../../app/assets/images/background.png",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "resizeMode": "repeat",
              "source": Object {
                "testUri": "../../../app/assets/images/background.png",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Image
            source={
              Object {
                "testUri": "../../../app/assets/images/ralph.png",
              }
            }
            style={
              Object {
                "alignSelf": "flex-end",
                "flex": 1,
              }
            }
          />,
          "style": Object {
            "flex": 1,
            "flexDirection": "row",
            "position": "absolute",
            "right": -50,
            "zIndex": 1,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": Object {
              "testUri": "../../../app/assets/images/ralph.png",
            },
            "style": Object {
              "alignSelf": "flex-end",
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(ViewNB)
            style={
              Object {
                "flex": 1,
                "zIndex": 2,
              }
            }
          >
            <Component>
              Hello
            </Component>
          </Styled(ViewNB)>,
          <Styled(ViewNB)
            style={
              Object {
                "flex": 1,
                "position": "absolute",
                "zIndex": 0,
              }
            }
          >
            <Image
              resizeMode="repeat"
              source={
                Object {
                  "testUri": "../../../app/assets/images/background.png",
                }
              }
            />
            <Image
              resizeMode="repeat"
              source={
                Object {
                  "testUri": "../../../app/assets/images/background.png",
                }
              }
            />
            <Image
              resizeMode="repeat"
              source={
                Object {
                  "testUri": "../../../app/assets/images/background.png",
                }
              }
            />
          </Styled(ViewNB)>,
          <Styled(ViewNB)
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "position": "absolute",
                "right": -50,
                "zIndex": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../app/assets/images/ralph.png",
                }
              }
              style={
                Object {
                  "alignSelf": "flex-end",
                  "flex": 1,
                }
              }
            />
          </Styled(ViewNB)>,
        ],
        "style": Object {
          "alignItems": "stretch",
          "flex": 1,
          "flexDirection": "column-reverse",
          "justifyContent": "flex-start",
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Component>
              Hello
            </Component>,
            "style": Object {
              "flex": 1,
              "zIndex": 2,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Hello",
            },
            "ref": null,
            "rendered": "Hello",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Image
                resizeMode="repeat"
                source={
                  Object {
                    "testUri": "../../../app/assets/images/background.png",
                  }
                }
              />,
              <Image
                resizeMode="repeat"
                source={
                  Object {
                    "testUri": "../../../app/assets/images/background.png",
                  }
                }
              />,
              <Image
                resizeMode="repeat"
                source={
                  Object {
                    "testUri": "../../../app/assets/images/background.png",
                  }
                }
              />,
            ],
            "style": Object {
              "flex": 1,
              "position": "absolute",
              "zIndex": 0,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "resizeMode": "repeat",
                "source": Object {
                  "testUri": "../../../app/assets/images/background.png",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "resizeMode": "repeat",
                "source": Object {
                  "testUri": "../../../app/assets/images/background.png",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "resizeMode": "repeat",
                "source": Object {
                  "testUri": "../../../app/assets/images/background.png",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Image
              source={
                Object {
                  "testUri": "../../../app/assets/images/ralph.png",
                }
              }
              style={
                Object {
                  "alignSelf": "flex-end",
                  "flex": 1,
                }
              }
            />,
            "style": Object {
              "flex": 1,
              "flexDirection": "row",
              "position": "absolute",
              "right": -50,
              "zIndex": 1,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": Object {
                "testUri": "../../../app/assets/images/ralph.png",
              },
              "style": Object {
                "alignSelf": "flex-end",
                "flex": 1,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
